### PR TITLE
Keep yessql.db as the fallback name for sqlite

### LINKED ISSUE #15154: Keep `yessql.db` as the fallback name for sqlite for backward compatability
@hishamco PR #7446 introduced a breaking change. All of my local tenants that use Sqlite are failing because they are expecting a file called `yessql.db` and not the app is looking for `OrchardCore.db`.

Here is what we should do. When creating new sites that use Sqllite, save `"DatabaseName": "OrchardCore.db",` to the shellSettings. If the shell-settings does not have `DatabaseName` value, then use `yessql.db`. This way we keep it backward compatible and only change it for new tenants. Ping me when you get this fix

### PR DESCRIPTION
**Pull request by MikeAlhayek**
@hishamco @Skrypt 

Fix #15154

### CODE REVIEW

**Comment by MikeAlhayek** [comment]
@hishamco note here I am using `orchardcore.db` as per suggestion from @sebastienros during meeting today.

**Comment by Skrypt** [comment]
I will leave that one for @sebastienros to decide :smile:

**Comment by hishamco** [comment]
> @hishamco note here I am using orchardcore.db as per suggestion from @sebastienros during meeting today.

Usually, I see the recording and then react to feedback suggestions, but you didn't mention that's the meeting suggestion feel free to close my PR

**Comment by MikeAlhayek** [comment]
The meeting suggestion to lower case the name. This PR provides the fix I described in the issue.

**Comment by hishamco** [comment]
There are a lot of changes in your PR that might not be necessary, I agree with @Skrypt that we might need some sort of logging or validation

Again while the IO check happened once I don't think it's a big issue

**Comment by MikeAlhayek** [comment]
@hishamco can you give me an example of an unnesasry change? This PR make renaming the file seemless without any breaking change. "Breaking change that we brought to main already".

I still don't think we should be doing the IO check if we don't have to. Validation should not be needed. We also don't validate if someone changes other appsettings like site name, or database prefix...  if you have access to the appsettings and make direct changes, we should assume that you know what you are doing and not add validation for that.

**Comment by Skrypt** [comment]
To be simpler. Test it, Rename the appsetting databasename and see what the app logs or not.

**Comment by hishamco** [comment]
According to the documentation, this should fix the issue:

```json
{
    "OrchardCore_Data_Sqlite": {
        "DatabaseName": "yessql.db",
        "UseConnectionPooling": false
    }
}
```

**Comment by MikeAlhayek** [comment]
@Skrypt if the file is missing, yessql will throw exception. That is enough for someone to know they screwed up the appsettings. I still don't believe we should have a check. In the framework if we check for everything IT does, we'll just bog down the framework.

Perhapse, we should just undo that PR that introduced the breaking change and keep the file called `yessql.db` and we don't need to worry about customizing the name :)

**Comment by MikeAlhayek** [comment]
@hishamco the problem is that it is a breaking change that could be avoided.

**Comment by hishamco** [comment]
You can if you set the settings as I mentioned in https://github.com/OrchardCMS/OrchardCore/pull/15157#issuecomment-1908445772

**Comment by Skrypt** [comment]
And the worst part is that it created a new OrchardCore.db file in my tenant folder...

**Comment by MikeAlhayek** [comment]
@Skrypt if you tested this PR, then it assumes that your file is yessql.db. If the file is called anything else then it should be in the settings file. 

In the settings file you can have 
{
   "DatabaseName": "jasmin.db"
}

and the file should be `jasmin.db`. If you don't specify anything then it would be yessql.db.  if you create a new site, then the file will be called orchardcore.db and the same name will be found in the settings.

**Comment by Skrypt** [comment]
All my point is about making a typo on that DatabaseName since the beginning. I tested the use case and I'm reporting you the issue. You said it would log ;it doesn't and it creates a new database named as what I'm adding in the appsettings.json file of my tenant instead of just throwing at the right place before creating a new database that will try to initialize a new setup. The worst part is that it doesn't even get me to the setup page. I get a 404 ... that's not good ... we can do better.

Don't want to argue more on this.

File: src/OrchardCore/OrchardCore.Data.YesSql/Options/SqliteOptions.cs [hunk 1]
```diff
@@ -17,8 +17,13 @@ public class SqliteOptions
         public bool UseConnectionPooling { get; set; } = true;
 
         /// <summary>
-        /// Gets or sets the database name. Defaults to <c>OrchardCore.db</c>.
+        /// Gets the database name.
         /// </summary>
-        public string DatabaseName { get; set; } = "OrchardCore.db";
+        public string DatabaseName { get; private set; }
```
Comment (sebastienros) [inline src/OrchardCore/OrchardCore.Data.YesSql/Options/SqliteOptions.cs:22]: get; set;

**Comment by sebastienros** [comment]
Please revert the PR that changed the database name. We'll put it back when the migration path is well-defined and the implementation is not brittle.

**Pull request closed without merging**